substrate/network-libp2p/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.?
+
+#![type_length_limit = "268435456"]
```

Comment:
long type!

Reply:
why doesn't `impl Trait` just assign unique numbers or something so we don't have to deal with this?

Reply:
I've added some `Box`es after I realized that the compile time was starting to be really long, so the limit might not need to be that high. However I don't see a reason to lower it either.